pykrige/rk.py

```diff
@@ -43,19 +79,55 @@ def __init__(
         weight=False,
         n_closest_points=10,
         verbose=False,
+        variogram_parameters=None,
+        variogram_function=None,
+        anisotropy_scaling=1.0,
+        anisotropy_angle=0.0,
+        enable_statistics=False,
+        coordinates_type="euclidean",
+        anisotropy_scaling_y=1.0,
+        anisotropy_scaling_z=1.0,
+        anisotropy_angle_x=0.0,
+        anisotropy_angle_y=0.0,
+        anisotropy_angle_z=0.0,
+        drift_terms=None,
+        point_drift=None,
+        external_drift=None,
+        external_drift_x=None,
+        external_drift_y=None,
+        functional_drift=None,
     ):
-
         validate_method(method)
         self.variogram_model = variogram_model
-        self.verbose = verbose
+        self.variogram_parameters = variogram_parameters
+        self.variogram_function = variogram_function
         self.nlags = nlags
         self.weight = weight
+        self.verbose = verbose
+        self.anisotropy_scaling = anisotropy_scaling
+        self.anisotropy_angle = anisotropy_angle
+        self.enable_statistics = enable_statistics
+        self.coordinates_type = coordinates_type
+        self.anisotropy_scaling_y = anisotropy_scaling_y
+        self.anisotropy_scaling_z = anisotropy_scaling_z
+        self.anisotropy_angle_x = anisotropy_angle_x
+        self.anisotropy_angle_y = anisotropy_angle_y
+        self.anisotropy_angle_z = anisotropy_angle_z
+        self.drift_terms = drift_terms
+        self.point_drift = point_drift
+        self.external_drift = external_drift
+        self.external_drift_x = external_drift_x
+        self.external_drift_y = external_drift_y
+        self.functional_drift = functional_drift
         self.model = None  # not trained
         self.n_closest_points = n_closest_points
         self.method = method
+        self.val_kw = "val" if self.method in threed_krige else "z"
```

Comment:
This would technically fail scikit-learn's validation check in [`check_no_attributes_set_in_init`](https://github.com/scikit-learn/scikit-learn/blob/fcba0d6c0b7a6af666d6fc0978b849251b53ebcc/sklearn/utils/estimator_checks.py#L2425). It might be better to set it in fit, and also maybe make it private `_val_kw` if users are not expected to interact with it?

Reply:
Will do so.